mapboxgl/templates/chloropleth.html

```diff
@@ -0,0 +1,83 @@
+{% extends "main.html" %}
+
+{% block javascript %}
+var legend = document.getElementById('legend');
+
+mapboxgl.accessToken = '{{ accessToken }}';
+
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: '{{ styleUrl }}', 
+    center: {{ center }},
+    zoom: {{ zoom }},
+    transformRequest: (url, resourceType)=> {
+        return {
+           url: [url.slice(0, url.indexOf("?")+1), "pluginName=PythonMapboxgl&", url.slice(url.indexOf("?")+1)].join('')
+         }
+    }
+});
+
+map.addControl(new mapboxgl.NavigationControl());
+
+calcCircleColorLegend({{ colorStops }}, "{{ colorProperty }}");
+
+map.on('style.load', function() {
+    
+    // Add data source
+    map.addSource("data", {
```

Comment:
What if the choropleth source is a vector tile?  This would be very common when working with Pandas, for example, when you have a tabular set of data that has a geo identifier like `postcode` that you want to match to a vector tile polygon source with a feature property matching `postcode`.  https://www.mapbox.com/mapbox-gl-js/example/data-join/

Reply:
I'd love to figure out a way to make a `choropleth` viz in `mapboxgl` use the same python class constructor whether a user has a local geojson file or has a vector tile source to match to.